specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json

```diff
@@ -3795,7 +3934,156 @@
       ],
       "description": "Data connector.",
       "discriminator": "kind",
-      "type": "object"
+      "type": "object",
+      "required": [
+        "kind"
+      ]
+    },
+    "DataConnectorsCheckRequirements": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/DataConnectorKind"
+        }
+      ],
+      "description": "Data connector requirements properties."
+    },
+    "DataConnectorAuthorizationModel": {
+      "description": "A model describing the authorization state and requirements missing (if there are any) for this connector",
+      "properties": {
+        "state": {
+          "description": "Describes the state of user's authorization for a connector kind.",
+          "$ref": "#/definitions/DataConnectorAuthorizationState"
+        },
+        "requiresAny": {
+          "description": "A list indicating which authorizations are needed for this connector (at least one), if there's no proper authorization.",
+          "items": {
+            "$ref": "#/definitions/DataConnectorAuthorizationType"
+          },
+          "type": "array"
+        }
+      }
+    },
+    "DataConnectorAuthorizationState": {
+      "description": "Describes the state of user's authorization for a connector kind.",
+      "enum": [
+        "Valid",
+        "Invalid"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "DataConnectorAuthorizationState",
+        "values": [
+          {
+            "value": "Valid"
+          },
+          {
+            "value": "Invalid"
+          }
+        ]
+      }
+    },
+    "DataConnectorAuthorizationType": {
+      "description": "Describes the types of user's authorizations.",
+      "enum": [
+        "GlobalAdmin",
+        "SecurityAdmin",
+        "TenantOnboardedToWorkspace",
+        "SecurityReader",
+        "ReadSubscriptionSecurityData"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "modelAsString": true,
+        "name": "DataConnectorAuthorizationType",
+        "values": [
```

Comment:
SDK automation tests failed because the values in `x-ms-enum` do not match with `enum` values above.

Reply:
Ok I fixed that now